app/uk/gov/hmrc/incometaxproperty/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -44,4 +44,23 @@ class IntegrationFrameworkConnector @Inject()(httpClient: HttpClient, appConf: A
   private def callGetBusinessDetails(url: URL)(implicit hc: HeaderCarrier): Future[GetBusinessDetailsResponse] = {
     httpClient.GET[GetBusinessDetailsResponse](url)
   }
+
+  def getPeriodicSubmission(taxYear: String, taxableEntityId: String, incomeSourceId: String)
+                           (implicit hc: HeaderCarrier): Future[Either[ApiError, PeriodicSubmissionModel]] = {
+    val apiVersion = "1649"
```

Comment:
2 things:

1.  Looking at the URL you are calling, this is not the implementation of `1649`. This is `1954`.
2. We are missing here the implementation of the API `1649`. 
3. And after having both API calls, in the service we need to check if the tax year is >23-24 onwards, to make one connector call or the other. We have examples of this in state benefits too.

Reply:
Both are there. Thanks